lib/spack/spack/test/cmd/install.py

```diff
@@ -384,7 +384,9 @@ def test_extra_files_are_archived(mock_packages, mock_archive, mock_fetch,
     archive_dir = os.path.join(
         spack.store.layout.metadata_path(s), 'archived-files'
     )
-    config_log = os.path.join(archive_dir, 'config.log')
+    config_log = os.path.join(archive_dir,
+                              mock_archive.repo_name,
```

Comment:
I have a mild preference for copying archive files relative to `Stage.source_path` vs. `Stage.path`, since in the latter case the name of the expanded directory influences the base directory of the archived files (and this isn't always as informative as the archive name itself).

Reply:
@scheibelp I'm having trouble wrapping my head around this: if the build directory is no longer in the source tree (the whole point of this PR), then unless the archive base is changed to `Stage.path` from `Stage.source_path`, then archived files from the build area (such as `CMakeCache.txt`, for instance, will no longer be allowed. Can we discuss this on the call today?